apkid/rules/elf/obfuscators.yara

```diff
@@ -215,3 +215,42 @@ rule avdobfuscator : obfuscator
     any of them and is_elf
 }
 
+rule arxan_native_arm : obfuscator
+{
+    meta:
+        description = "Arxan"
+        url         = "https://www.arxan.com/resources/technology/app-code-obfuscation"
+        sample      = "5bbb241d41c4150798b5800e62afcb6f49e05755d2fd89c7a9f7e356609c9012"
+        author      = "Eduardo Novella"
+
+    strings:
+        $a = {
+            00 10 90 E5    // LDR R1, [R0]
+            00 00 81 E0    // ADD R0, R1, R0
+            03 10 00 E0    // AND R1, R0, R3
+            02 20 A0 E3    // MOV R2, #2
+            92 01 01 E0    // MUL R1, R2, R1
+            03 00 20 E0    // EOR R0, R0, R3
+            01 00 80 E0    // ADD R0, R0, R1
+            00 F0 A0 E1    // MOV PC, R0
+        }
+
+        $b = {
+            4F F0 01 00    // MOV.W   R0, #1
+            02 A1          // ADR     R1, loc_191658
+            01 FB 00 F0    // MUL.W   R0, R1, R0
+            87 46          // MOV     PC, R0
+        }
+
+        $c = {
+            ?? ?? ?? E?
+            91 00 00 E0    // MUL     R0, R1, R0
+            00 F0 A0 E1    // MOV     PC, R0
+        }
+
+    condition:
+        // TODO: Count occurrences and > 20 times //(for 10 i in (1..#a): (@a[i] >= 50 and @a[i] <= 100))
+        // TODO: ARM64 & X86
+        ($a or $b or $c) and elf.machine == elf.EM_ARM
```

Comment:
I'm guessing you're doing a machine check because it's more specific than is_elf and might be a little faster, yes?

Reply:
It's indeed only matching arm32. Feel free to suggest any improvement to it.

Reply:
Sorry if I wasn't explicit -- it's pretty clear it's only matching a specific machine type; my question was "why". It's fine if you don't have a strong answer, but my intention is only to probe and learn more.

Reply:
Ok, I guess I can't force you to engage in conversation. 🤔

Reply:
why not? Of course yes.... So many things going on and sometimes miss comments